pkg/build/controller/build/defaults/defaults_test.go

```diff
@@ -178,14 +213,71 @@ func TestGlobalProxyDefaults(t *testing.T) {
 		}
 	}
 
+	// docker builds should have the defaulted env vars applied to the build pod
+	pod = testutil.Pod().WithBuild(t, testutil.Build().WithDockerStrategy().AsBuild())
+	err = admitter.ApplyDefaults((*corev1.Pod)(pod))
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	httpProxyFound, httpsProxyFound, noProxyFound = false, false, false
+	for _, ev := range pod.Spec.Containers[0].Env {
+		if ev.Name == "HTTP_PROXY" {
+			if ev.Value != "http" {
+				t.Errorf("unexpected value %s", ev.Value)
+			}
+			httpProxyFound = true
+		}
+		if ev.Name == "HTTPS_PROXY" {
+			if ev.Value != "https" {
+				t.Errorf("unexpected value %s", ev.Value)
+			}
+			httpsProxyFound = true
+		}
+		if ev.Name == "NO_PROXY" {
+			if ev.Value != "no" {
+				t.Errorf("unexpected value %s", ev.Value)
+			}
+			noProxyFound = true
+		}
+	}
+	if !httpProxyFound {
+		t.Errorf("HTTP_PROXY not found")
+	}
+	if !httpsProxyFound {
+		t.Errorf("HTTPS_PROXY not found")
+	}
+	if !noProxyFound {
+		t.Errorf("NO_PROXY not found")
+	}
+
+	// docker builds should not have the defaulted env vars applied to the build
```

Comment:
is there a reason you don't do this check for s2i?

Reply:
is that not what I am doing below this line? (might have to expand to view the existing code) 
https://github.com/openshift/origin/pull/22071/files#diff-d1abaf8fe636c39884e483bbb6aa08d6R197

Reply:
yeah, sorry that's what i was looking for. thanks.

Reply:
np, it was a bit hidden